File: JAVA/916. Word Subsets.java
class Solution {
    public List<String> wordSubsets(String[] w, String[] v) {
        Map<Character, Integer> r = new HashMap<>();
        List<String> a = new ArrayList<>();
        for (String s : v) {
            Map<Character, Integer> m = new HashMap<>();
            for (char c : s.toCharArray()) {
                m.put(c, m.getOrDefault(c, 0) + 1);
                r.put(c, Math.max(r.getOrDefault(c, 0), m.get(c)));
            }
        }
        for (String s : w) {
            Map<Character, Integer> l = new HashMap<>();
            for (char c : s.toCharArray()) {
                l.put(c, l.getOrDefault(c, 0) + 1);
            }
            boolean t = true;
            for (Map.Entry<Character, Integer> entry : r.entrySet()) {
                if (entry.getValue() > l.getOrDefault(entry.getKey(), 0)) {
                    t = false;
                    break;
                }
            }
            if (t) {
                a.add(s);
            }
        }
        return a;
    }
}

File: C++/916. Word Subsets.cpp
class Solution {
public:
    vector<string> wordSubsets(vector<string>& w, vector<string>& v) {
        map<char,int>r;
        vector<string>a;
        for(string s:v){
            map<char,int>m;
            for(char c:s){
                m[c]++;
                r[c]=max(r[c],m[c]);
            }
        }
        for(string s:w){
            map<char,int>l;
            for(char c:s)
                l[c]++;
            bool t=true;
            for(auto i=r.begin();i!=r.end();i++){
                if(i->second >l[i->first]){
                    t=false;
                    break;
                }
            }
            if(t)
                a.push_back(s);
        }
        return a;
    }
};